src/components/encoding-pane/encoding-shelf.tsx

```diff
@@ -72,7 +72,8 @@ class EncodingShelfBase extends React.PureComponent<
             attachment="top left"
             targetAttachment="bottom left"
           >
-            {(fieldDef && !isWildcardChannelId(id)) ?
+            {(fieldDef && !isWildcardChannelId(id) && (this.props.id.channel === 'x' ||
```

Comment:
Disable the drop-down for channels that are not positional, since property-editor is not supported for those channels yet.

Reply:
Why commenting in the PR?  If you have to comment in a PR, that's a sign that you better add comment to the code? 

(Commenting in the PR is better for asking me questions.)

Reply:
you can refer to `id` without `this.props` too.